packages/components/bucket-policy/bucket-policy-body.tsx

```diff
@@ -0,0 +1,442 @@
+    });
+  };
+
+/*const onChangeReplication: onChangeReplicationSelectProps =
+  (replicationObj, dispatch) => (resource) => {
+    const alreadyIncluded = replicationObj.some((obj) => obj.obcName === resource?.metadata?.name);
+    const payload: ReplicationOBC[] = alreadyIncluded
+      ? replicationObj.filter((obj) => obj.obcName !== resource?.metadata?.name)
+      : [...replicationObj, {
+        obcName: resource?.metadata?.name,
+        targetBucket: resource?.spec?.bucketName,
+      }];
+    dispatch({
+      type: BucketPolicyActionType.SET_REPLICATION_OBC,
+      payload,
+    })
+  };*/
+
+const DataResourceDropdown: React.FC<DataResourceDropdownProps> = ({
+  data,
+  onChange,
+  launchModal,
+  disableAllExcept,
+  variant = SelectVariant.single,
+  ...props
+}) => {
+  const { t } = useTranslation();
+  const [isOpen, setOpen] = React.useState(false);
+
+  const dropdownOptions: JSX.Element[] = React.useMemo(
+    () =>
+      data.reduce(
+        (acc, dataResource) => {
+          const name = getName(dataResource);
+          const isDisabled =
+            (!_.isEmpty(disableAllExcept) &&
+              !disableAllExcept?.some((itemName) => itemName === name)) ||
+            dataResource?.status?.phase !== 'Ready';
+          acc.push(
+            <SelectOption
+              key={getUID(dataResource)}
+              value={name}
+              isDisabled={isDisabled}
+              data-test={`${name}-dropdown-item`}
+            />
+          );
+          return acc;
+        },
+        [
+          <Button
+            key="data-source-add-modal"
+            className="create-data-source__modal-button"
+            variant="plain"
+            onClick={() => {
+              setOpen(false);
+              launchModal();
+            }}
+          >
+            {t('Add new data source')}
+          </Button>,
+          <Divider component="li" key={'data-source-divider'} />,
+        ]
+      ),
+    [t, data, disableAllExcept, launchModal]
+  );
+
+  return (
+    <Select
+      {...props}
+      className={classNames(props?.className)}
+      variant={variant}
+      aria-label={t('Select data source')}
+      onToggle={setOpen}
+      onSelect={(e, selection) => {
+        onChange(selection as string);
+        variant !== SelectVariant.checkbox && setOpen(false);
+      }}
+      isOpen={isOpen}
+      id="data-source-dropdown"
+      placeholderText={t('Select options')}
+      noResultsFoundText={t('No results found')}
+    >
+      {dropdownOptions}
+    </Select>
+  );
+};
+
+const SingleDataResource: React.FC<DataResourceProps> = ({
+  state,
+  dispatch,
+  launchModal,
+  ...props
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <FormGroup
+      fieldId="read-write-data-source"
+      className="data-source-type__form--margin"
+      label={t('Select a read and write data source')}
+    >
+      <DataResourceDropdown
+        {...props}
+        selections={state.readWriteSingle}
```

Comment:
why is this prop needed for the dropdown ? i can't see it being used in the dropdown

Reply:
I will take a look once, but I think it is been used... it is just whatever we are receiving we are passing it down... that is it...